### ISSUE TITLE
[Bug]: kill update and delete sql reported panic error

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#7615]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.7.0 or 8b23a93):99d3fa4232768ec577284e9992221fc0dce30d9e
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

kill sql update table_100w set clo2=3000 where clo1>50 reported panic error, plz see steps to Reproduce
![image](https://user-images.githubusercontent.com/113406637/212227844-d5982b97-6d32-454a-8c27-3a556e7556e9.png)
![image](https://user-images.githubusercontent.com/113406637/212227353-cf736464-a780-4454-86a2-3d6b56d6641f.png)
![image](https://user-images.githubusercontent.com/113406637/212227879-7e51372d-bdfd-41f1-8458-64c47367048e.png)
and I try  after insert ,execute update table_100w set clo2=3000 where clo1>50, kill success
![image](https://user-images.githubusercontent.com/113406637/212234369-2591f47d-e55a-473e-b1d3-e395a4a77419.png)

error log:
2023/01/13 10:54:22.639009 +0800 ERROR frontend/util.go:510 internal error: panic runtime error: index out of range [0] with length 0:
runtime.goPanicIndex
        /opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/panic.go:113
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.getZonemapDataFromMeta
        /Users/heni/test-envir/matrixone/pkg/vm/engine/disttae/util.go:137
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.needRead
        /Users/heni/test-envir/matrixone/pkg/vm/engine/disttae/txn.go:717
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*table).Ranges
        /Users/heni/test-envir/matrixone/pkg/vm/engine/disttae/table.go:222
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).generateNodes
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:1444
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compileTableScan
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:690
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compilePlanScope
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:445
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compilePlanScope
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:462
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compilePlanScope
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:568
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compileQuery
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:288
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).compileScope
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:208
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).Compile
        /Users/heni/test-envir/matrixone/pkg/sql/compile/compile.go:104
github.com/matrixorigin/matrixone/pkg/frontend.(*TxnComputationWrapper).Compile
        /Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:2397
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery
        /Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:3622
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest
        /Users/heni/test-envir/matrixone/pkg/frontend/mysql_cmd_executor.go:4153
github.com/matrixorigin/matrixone/pkg/frontend.(*Routine).handleRequest
        /Users/heni/test-envir/matrixone/pkg/frontend/routine.go:165
github.com/matrixorigin/matrixone/pkg/frontend.(*RoutineManager).Handler
        /Users/heni/test-envir/matrixone/pkg/frontend/routine_manager.go:290
github.com/fagongzi/goetty/v2.(*server).doConnection
        /Users/heni/go/pkg/mod/github.com/fagongzi/goetty/v2@v2.0.3-0.20221212132037-abf2d4c05484/application.go:381
github.com/fagongzi/goetty/v2.(*server).doStart.func2.1
        /Users/heni/go/pkg/mod/github.com/fagongzi/goetty/v2@v2.0.3-0.20221212132037-abf2d4c05484/application.go:344

### Steps to Reproduce
create external table ex_table_100w(clo1 tinyint,clo2 smallint,clo3 int,clo4 bigint,clo5 tinyint unsigned,clo6 smallint unsigned,clo7 int unsigned,clo8 bigint unsigned,col9 float,col10 double,col11 varchar(255),col12 Date,
col13 DateTime,col14 timestamp,col15 bool,col16 decimal(5,2),col17 text,col18 varchar(255),col19 varchar(255),col20 varchar(255))infile{"filepath"='/Users/heni/test_data/1000000_20_columns_load_data.csv'} ;
create table table_100w(clo1 tinyint,clo2 smallint,clo3 int,clo4 bigint,clo5 tinyint unsigned,clo6 smallint unsigned,clo7 int unsigned,clo8 bigint unsigned,col9 float,col10 double,col11 varchar(255),col12 Date, col13 DateTime,col14 timestamp,col15 bool,col16 decimal(5,2),col17 text,col18 varchar(255),col19 varchar(255),col20 varchar(255));
insert into table_100w select * from  ex_table_100w;
update table_100w set clo2=3000 where clo1>80;
update table_100w set clo2=3000 where clo1>50;

**Comment by heni02** [comment]
mo log : [log.txt](https://github.com/matrixorigin/matrixone/files/10407887/log.txt)

**Comment by heni02** [comment]
same problem as kill delete sql
mysql> delete from table_100w where clo1>20;
ERROR 20101 (HY000): internal error: panic runtime error: index out of range [0] with length 0:
runtime.goPanicIndex
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/panic.go:113
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.getZonemapDataFromMeta
	/Users/heni/test-envir/matrixone/pkg/vm/engine/disttae/util.go:137
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.needRead
	/Users/heni/test-envir/matrixone/pkg/vm/engine/disttae/txn.go:717
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*table).Range

**Comment by daviszhen** [comment]
the context.Conetxt has been canceled. the computation engine behaves abnormally. something is related to the https://github.com/matrixorigin/matrixone/issues/6639

**Comment by daviszhen** [comment]
I have reproduced it.

```

mysql> update table_100w set clo2=3000 where clo1>50;
ERROR 20101 (HY000): internal error: panic runtime error: index out of range [0] with length 0: 
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:113
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.getZonemapDataFromMeta
	/Users/pengzhen/Documents/GitHub/matrixone/pkg/vm/engine/disttae/util.go:137
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.needRead
	/Users/pengzhen/Documents/GitHub/matrixone/pkg/vm/engine/disttae/txn.go:717
github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*table).Ranges
	/U
mysql> 
```

**Comment by m-schen** [comment]
who develop the KILL？

**Comment by nnsgmsone** [comment]
who develop the KILL？

**Comment by nnsgmsone** [comment]
The kill doesn't actually kill the query, it needs to be modified by Peng Zheng

**Comment by nnsgmsone** [comment]
The reason for this is that the wrong parameters were passed to read the metadata, the prefetch was passed, so when the read data was cancelled, it didn't report the error correctly.

**Comment by nnsgmsone** [comment]
fixed by #7959

**Comment by heni02** [comment]
confirm,closed
commit id:346277e3cb8e7aec9d68c043a3e9249e2c3cb842
![image](https://user-images.githubusercontent.com/113406637/218296887-c733fcc1-71b9-4fe1-b40b-f7d3bbc0d58f.png)
![image](https://user-images.githubusercontent.com/113406637/218296903-0c40b311-0703-4f2f-be75-25738a88ced3.png)
